tests/OTAnalytics/application/use_cases/test_load_otconfig.py

```diff
@@ -0,0 +1,92 @@
+from datetime import datetime
+from unittest.mock import MagicMock, Mock
+
+import pytest
+
+from OTAnalytics.application.parser.config_parser import OtConfig
+from OTAnalytics.application.state import ConfigurationFile
+from OTAnalytics.application.use_cases.load_otconfig import (
+    LoadOtconfig,
+    UnableToLoadOtconfigFile,
+)
+from OTAnalytics.application.use_cases.section_repository import SectionAlreadyExists
+
+
+class TestLoadOtconfig:
+    @pytest.fixture
+    def otconfig(self) -> OtConfig:
+        project = Mock()
+        project.name = "my project"
+        project.start_date = datetime(2021, 1, 1)
+
+        otconfig = Mock()
+        otconfig.project = project
+        otconfig.videos = Mock()
+        otconfig.sections = Mock()
+        otconfig.flows = Mock()
+        return otconfig
+
+    def test_load(self, otconfig: OtConfig) -> None:
+        clear_repositories = Mock()
+        config_parser = Mock()
+        config_parser.parse.return_value = otconfig
+
+        update_project = Mock()
+        add_videos = Mock()
+        add_sections = Mock()
+        add_flows = Mock()
+        deserialization_result = Mock()
+        deserializer = Mock()
+        deserializer.return_value = deserialization_result
+
+        load_otconfig = LoadOtconfig(
+            clear_repositories,
+            config_parser,
+            update_project,
+            add_videos,
+            add_sections,
+            add_flows,
+            deserializer,
+        )
+        observer = Mock()
+        load_otconfig.register(observer)
+
+        file = Mock()
+        load_otconfig.load(file)
+
+        update_project.assert_called_once_with(
+            otconfig.project.name, otconfig.project.start_date
+        )
+        add_videos.add.assert_called_once_with(otconfig.videos)
+        add_sections.add.assert_called_once_with(otconfig.sections)
+        add_flows.add.assert_called_once_with(otconfig.flows)
+        observer.assert_called_once_with(
+            ConfigurationFile(file, deserialization_result)
+        )
+
+    def test_load_error(self, otconfig: OtConfig) -> None:
+        clear_repositories = Mock()
+        config_parser = Mock()
+        config_parser.parse.return_value = otconfig
+
+        add_sections = MagicMock()
+        add_sections.add.side_effect = SectionAlreadyExists
+
+        load_otconfig = LoadOtconfig(
+            clear_repositories,
+            config_parser,
+            Mock(),
+            Mock(),
+            add_sections,
+            Mock(),
+            Mock(),
+        )
+        observer = Mock()
+        load_otconfig.register(observer)
+        file = Mock()
+        with pytest.raises(UnableToLoadOtconfigFile):
+            load_otconfig.load(file)
+
+        assert clear_repositories.call_count == 2
```

Comment:
Why is the repository cleared twice?

Reply:
A `SectionAlreadyExists` or `FlowAlreadyExists` Exception will be thrown if  a section or flow already exists. This implies that the OTFlow or OTConfig file is not valid.
Since we are sequentially adding each section or flow to the repository, the repository could be only partly loaded when any of the above exceptions occur. To avoid this state, we decided to just clear the repositories again.

The correct behaviour would be to restore the state before a configuration file was loaded if an error occurs.
This would require validation logic in the `AddSections` or `AddFlows` use cases. But maybe that is actually the responsibility of the `FlowParser` or `ConfigFileParser` to return a validly parsed configuration. I leaning to the latter solution. @briemla

Reply:
The latter solution sounds good. Can you please file a feature / epic in OP?